WooCommerce/Classes/ViewRelated/Themes/ThemesPreviewView.swift

```diff
@@ -0,0 +1,207 @@
+
+                } else {
+                    errorView
+                }
+            }
+            .toolbar {
+                if let url = theme.themeThumbnailURL {
+                }
+                ToolbarItem(placement: .navigationBarLeading) {
+                    Image(uiImage: .closeButton)
+                }
+
+                ToolbarItem(placement: .navigationBarTrailing) {
+                    Menu {
+                        ForEach(PreviewDevice.allCases) { device in
+                          menuItem(for: device)
+                        }
+                    } label: {
+                         Image(systemName: "macbook.and.iphone")
+                            .bodyStyle()
+                    }
+                }
+            }
+            .navigationBarTitleDisplayMode(.inline)
+        }
+    }
+
+    private func menuItem(for device: PreviewDevice) -> some View {
+        Button {
+            self.selectedDevice = device
+        } label: {
+            Text(device.menuTitle)
+            if self.selectedDevice == device {
+                Image(systemName: "checkmark")
+            }
+        }
+    }
+}
+
+private extension ThemesPreviewView {
+    var errorView: some View {
+        VStack(spacing: Layout.contentPadding) {
+            Spacer()
+            Text(Localization.errorLoadingThemeDemo)
+                .secondaryBodyStyle()
+            Spacer()
+        }
+        .frame(maxWidth: .infinity)
+    }
+}
+
+private extension ThemesPreviewView {
+    private enum Layout {
+        static let toolbarPadding: CGFloat = 16
+        static let dividerHeight: CGFloat = 1
+        static let footerPadding: CGFloat = 16
+        static let contentPadding: CGFloat = 16
+    }
+
+    private enum Localization {
+        static let menuMobile = NSLocalizedString(
+            "themesPreviewView.menuMobile",
+            value: "Mobile",
+            comment: "Menu item: mobile"
+        )
+        static let menuTablet = NSLocalizedString(
+            "themesPreviewView.menuTablet",
+            value: "Tablet",
+            comment: "Menu item: tablet"
+        )
+
+        static let menuDesktop = NSLocalizedString(
+            "themesPreviewView.menuMobile",
+            value: "Desktop",
+            comment: "Menu item: desktop"
+        )
+        static let startWithThemeButton = NSLocalizedString(
+            "themesPreviewView.startWithThemeButton",
+            value: "Start with This Theme",
+            comment: "Button in theme preview screen to pick a theme."
+        )
+        static let themeName = NSLocalizedString(
+            "themesPreviewView.themeName",
+            value: "Theme: %@",
+            comment: "Name of the theme being previewed."
+        )
+
+        static let errorLoadingThemeDemo = NSLocalizedString(
+            "ThemesPreviewView.errorLoadingThemeDemo",
+            value: "This theme has no theme demo. Please try another theme.",
```

Comment:
How about "Unable to render the demo for this theme. Please try another theme."?

Reply:
My thought was this was related to the fact that in `WordPressTheme`, `demoURI` can be empty string, which will make `errorView` appear in this view. But your message makes more sense too, I think, as it's more generic and refers to the app itself instead of the theme.

Reply:
https://github.com/woocommerce/woocommerce-ios/pull/11452/commits/ac1101c0c98ecc774dcb8c5f967a9bebd5be163a